### PR TITLE
install: Use amd maintained base for mpi/ucx/...

### PR DESCRIPTION
**Pull request by mloubout**
To be tested but that should work (it build locally at least)

### CODE REVIEW

**Review by FabioLuporini — COMMENTED** [review]
I love this. Let me test it

**Comment by mloubout** [comment]
It gives me some weird permission error locally not sure why but yeah seem like the "nicer" way to do it without caring about maintaining a base

File: devito/arch/compiler.py [hunk 1]
```diff
@@ -436,10 +436,8 @@ def __init__(self, *args, **kwargs):
             # Add flags for OpenMP offloading
             if language in ['C', 'openmp']:
                 self.ldflags += ['-target', 'x86_64-pc-linux-gnu']
-                self.ldflags += ['-fopenmp',
-                                 '-fopenmp-targets=amdgcn-amd-amdhsa',
-                                 '-Xopenmp-target=amdgcn-amd-amdhsa']
-                self.ldflags += ['-march=%s' % platform.march]
+                self.ldflags += ['-fopenmp']
+                self.ldflags += ['--offload-arch=%s' % platform.march]
```
Comment (mloubout) [inline devito/arch/compiler.py:440]: @FabioLuporini

File: devito/arch/archinfo.py [hunk 2]
```diff
@@ -701,7 +701,7 @@ def march(cls):
         #     mygpu will only print values accepted by cuda clang in
         #     the clang argument --cuda-gpu-arch.
         try:
-            p1 = Popen(['mygpu', '-d', 'gfx900'], stdout=PIPE, stderr=PIPE)
+            p1 = Popen(['/opt/rocm-5.4.2/llvm/bin/offload-arch'], stdout=PIPE, stderr=PIPE)
```
Comment (mloubout) [inline devito/arch/archinfo.py:704]: @FabioLuporini

**Comment by mloubout** [comment]
Superseeded by #2104, closing

**Pull request closed without merging**